src/index.js

```diff
@@ -69,13 +69,12 @@ export default class AppCachePlugin {
     const {options: {output: outputOptions = {}} = {}} = compiler;
     const {publicPath = ''} = outputOptions;
 
-    compiler.plugin('emit', (compilation, callback) => {
```

Comment:
I'd like to maintain backwards compatibility here, if possible. For instance, we could detect whether `compiler.hooks.emit.tap` is available, but fall back to `compiler.plugin` if it isn't.